src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -115,12 +124,33 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
         }
 
         private void CodeCleanupOrFormat(
-            FormatDocumentCommandArgs args, Document document,
+            FormatDocumentCommandArgs args, Document document, ICodeCleanupService codeCleanupService,
             IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
-            var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
-
             var docOptions = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var workspace = document.Project.Solution.Workspace;
+
+            // AB Test
+            // If the feature is OFF and the feature options have not yet been set to their assigned flight, do so now
+            if (!docOptions.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting)
+                && !docOptions.GetOption(CodeCleanupABTestOptions.ParticipatedInExperiment))
+            {
+                var experimentationService = document.Project.Solution.Workspace.Services.GetService<IExperimentationService>();
+
+                if (experimentationService != null)
+                {
+                    if (experimentationService.IsExperimentEnabled(s_experimentName))
+                    {
+                        workspace.Options = workspace.Options.WithChangedOption(new OptionKey(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, document.Project.Language), true);
```

Comment:
what turns this off, once the user is not part of the experiment anymore?

Reply:
good. point. probably, it shouldn't save the option back